fml/task_runner_merger.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FML_SHELL_COMMON_TASK_RUNNER_MERGER_H_
+#define FML_SHELL_COMMON_TASK_RUNNER_MERGER_H_
+
+#include "flutter/fml/macros.h"
+#include "flutter/fml/memory/ref_counted.h"
+#include "flutter/fml/message_loop_task_queues.h"
+
+namespace fml {
+
+class MessageLoopImpl;
+
+class TaskRunnerMerger : public fml::RefCountedThreadSafe<TaskRunnerMerger> {
+ public:
+  void MergeWithLease(size_t lease_term);
+
+  void ExtendLease(size_t lease_term);
+
+  // Returns true if we unmerged this turn.
+  bool DecrementLease();
+
+  bool AreMerged() const;
+
+  TaskRunnerMerger(fml::TaskQueueId platform_queue_id,
+                   fml::TaskQueueId gpu_queue_id);
+
+  bool OnWrongThread();
```

Comment:
Coming back after reading the implementation - I guess this is checking whether the current thread is the current active GPU thread, maybe it will help to name this method `IsOnGpuThread` ?

And this method probably needs a comment anyway.

Reply:
this method is to answer the following question:

1. If we are merged are we on the GPUThread.
2. If we are unmerged are we on the PlatformThread.

I added a comment explaining the same. Changed the name to `IsOnRasterizingThread`